agent/util-scripts/pbench-tool-meister-stop

```diff
@@ -66,86 +90,120 @@ def main(argv):
-        logger.debug("tm_pids = %r", tm_pids)
+        group = cli_params.tool_group
+
+    sysinfo, bad_l = cli_verify_sysinfo(cli_params.sysinfo)
+    if bad_l:
+        logger.error('invalid sysinfo option(s), "{}"', ",".join(bad_l))
 
-    ret_val = 0
+    interrupt = cli_params.interrupt
+    if interrupt and sysinfo:
+        # Don't collect system information if the CLI invocation specified an
+        # interrupt occured.
+        logger.warning("system information not collected when --interrupt specified")
+        sysinfo = ""
 
-    logger.debug("terminating %d pids at localhost:%d", expected_pids, redis_port)
-    terminate_msg = dict(action="terminate", group=None, directory=None, args=None)
     try:
-        num_present = redis_server.publish(
-            channel, json.dumps(terminate_msg, sort_keys=True)
-        )
+        full_hostname = os.environ["_pbench_full_hostname"]
+        benchmark_run_dir = Path(os.environ["benchmark_run_dir"]).resolve(strict=True)
     except Exception:
-        logger.exception("Failed to publish terminate message")
-        ret_val = 1
-    else:
-        if num_present != expected_pids:
+        logger.exception("failed to fetch parameters from the environment")
+        return 1
+
+    try:
+        redis_server_pid_file = (
+            benchmark_run_dir / "tm" / f"redis_{redis_port:d}.pid"
+        ).resolve(strict=True)
+    except FileNotFoundError:
+        # No Redis server, nothing to do.
+        return 0
+
+    # The Redis server is always running on the local host with the CLI.
+    with Client(
+        redis_host="localhost",
+        redis_port=redis_port,
+        channel_prefix=cli_tm_channel_prefix,
+        logger=logger,
+    ) as client:
+        # First we end the persistent tools
+        tool_dir = benchmark_run_dir / f"tools-{group}"
+        try:
+            tool_dir.mkdir(exist_ok=True)
+        except Exception as exc:
             logger.error(
-                "Failed to terminate %d pids, only encountered %d on the channel",
-                expected_pids,
-                num_present,
+                'failed to create tool output directory, "{}": {}', tool_dir, exc
             )
+        else:
+            # FIXME - check return status.
+            client.publish(group, tool_dir, "end", None)
+        # Next we collect the system configuration information
+        if sysinfo:
+            sysinfo_path = benchmark_run_dir / "sysinfo" / "end"
+            try:
+                sysinfo_path.mkdir(parents=True)
+            except Exception:
+                logger.error(
+                    "Unable to create sysinfo-dump directory base path: {}",
+                    sysinfo_path,
+                )
+            else:
+                logger.info("Collecting system information")
+                info_log("Collecting system information")
+                # FIXME - check return status.
+                client.publish(group, sysinfo_path, "sysinfo", sysinfo)
```

Comment:
We might want to check for the return status to report the success or failure, perhaps in the logs, perhaps to the console on failure, but we don't gate the success/failure of the `pbench-tool-meister-stop` on that.

Reply:
The `publish` method already logs error messages, so we should just remove the `FIXME`.